community/spatial-index/src/test/java/org/neo4j/gis/spatial/index/EnvelopeTest.java

```diff
@@ -106,6 +107,49 @@ public void shouldHandleIntersectionsIn2D()
         testOverlaps(left, new Envelope( 0.5, 1.5, 0.0, 1.0 ), true, 0.5, 0.5);        // overlaps right half
     }
 
+    @Test
+    public void testWithSideRatioNotTooSmall2D()
+    {
+        // No change expected
+        double[] from = new double[] {0, 0};
+        double[] to = new double[] {1, 1};
+
+        Envelope envelope = new Envelope( from, to ).withSideRatioNotTooSmall();
+        double[] expectedFrom = new double[] {0, 0};
+        double[] expectedTo = new double[] {1, 1};
+        assertArrayEquals( expectedFrom, envelope.min , 0);
+        assertArrayEquals( expectedTo, envelope.max, 0 );
+
+        // Expected to change
+        to = new double[] {100, 0.1};
+        Envelope envelope2 = new Envelope( from, to ).withSideRatioNotTooSmall();
+        double[] expectedTo2 = new double[] {100, 1};
+        assertArrayEquals( expectedFrom, envelope2.min , 0);
+        assertArrayEquals( expectedTo2, envelope2.max, 0.00001 );
+    }
+
+    // Works for any number of dimensions, and 4 is more interesting than 3
+    @Test
+    public void testWithSideRatioNotTooSmall4D()
+    {
+        // No change expected
+        double[] from = new double[] {0, 0, 0, 0};
+        double[] to = new double[] {1, 1, 1, 1};
+
+        Envelope envelope = new Envelope( from, to ).withSideRatioNotTooSmall();
+        double[] expectedFrom = new double[] {0, 0, 0, 0};
+        double[] expectedTo = new double[] {1, 1, 1, 1};
+        assertArrayEquals( expectedFrom, envelope.min , 0);
+        assertArrayEquals( expectedTo, envelope.max, 0 );
+
+        // Expected to change
+        to = new double[] {100, 0.1, 12, 0.01};
+        Envelope envelope2 = new Envelope( from, to ).withSideRatioNotTooSmall();
+        double[] expectedTo2 = new double[] {100, 1, 12, 1};
```

Comment:
In both this and the test above all changed values are expected to become 1, due to the highest diff being 100. Maybe we should test with some other value as well.